### PR TITLE
TGP-1019: add endpoint to maintain a profile

### PR DESCRIPTION
**Pull request by william-eyre**
The validation on the EORI will be done as part of https://jira.tools.tax.service.gov.uk/browse/TGP-885

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/service/MaintainProfileService.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (pauloncast) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/service/MaintainProfileService.scala:1]: I think we have to abstract out a service for the rest of the endpoints too if we are doing for one, so as to have some consistency. And they can all extend a BaseService where shared functionalities like `logMessageAndReturnError` can reside

Comment (william-eyre) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/service/MaintainProfileService.scala:1]: Yeah I agree, but I will do that as a separate PR, I want to keep this one a bit smaller

Comment (pauloncast) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/service/MaintainProfileService.scala:1]: Alright then. Happy to resolve.

**Review by pauloncast — APPROVED** [review]
LGTM

**Pull request merged**